src/Background/estimators.jl

```diff
@@ -61,3 +61,50 @@ julia> estimate_background(Mode, data)
 struct Mode <: BackgroundEstimator end
 
 estimate_background(::Mode, data; dims = :) = mode(data)
+
+"""
+    SourceExtractor
+
+This estimator returns the background of the input using the SourceExtractor algorithm.
+
+The background is calculated using a mode estimator of the form `(2.5 * median) - (1.5 * mean)`.
+
+If `(mean - median) / std > 0.3` then the median is used and if `std = 0` then the mean is used.
+
+# Example
+```jldoctest
+julia> data = ones(5 ,5);
+
+julia> estimate_background(SourceExtractor, data)
+1.0
+
+julia> estimate_background(SourceExtractor, data, dims=1)
+1×5 Array{Float64,2}:
+ 1.0  1.0  1.0  1.0  1.0
+```
+"""
+struct SourceExtractor <: BackgroundEstimator end
+
+function estimate_background(::SourceExtractor, data; dims = :)
+    _mean = mean(data, dims = dims)
+    _median = median(data, dims = dims)
+    _std = std(data, dims = dims)
+
+    background = 2.5 * _median - 1.5 * _mean
+    if !(background isa Float64)
+        for i in length(background)
```

Comment:
Don't assume that indexes start from 1, use
```suggestion
        for i in eachindex(background)
```